app/search/query/base.py

```diff
@@ -0,0 +1,103 @@
+import bleach
+
+from memorize import memorize
+from memcached import memcached
+
+
+class BaseQueryEngine(object):
+    """
+    Query engines are subclasses of this that are used to fetch the top search
+    result for a search string.
+
+    Example: "the martian" =>
+
+    {
+        "abstract": "The <b>Martian</b> is a 2011 science fiction novel...",
+        "clickurl": "https://en.wikipedia.org/wiki/The_Martian_(Andy_Weir)",
+        "date": "",
```

Comment:
lol, the BOSS docs are a wee bit incomplete: https://developer.yahoo.com/boss/search/boss_api_guide/web.html

![](https://www.dropbox.com/s/pc5vhd9lfid622e/Screenshot%202015-12-31%2014.21.52.png?dl=0&raw=true)